server/models/Collective.ts

```diff
@@ -2845,7 +2845,11 @@ class Collective extends Model<
   };
 
   // get the host of the parent collective if any, or of this collective
-  getHostCollective = function () {
+  getHostCollective = async function ({ returnEvenIfNotApproved = false } = {}) {
+    if (!this.isActive && !returnEvenIfNotApproved) {
```

Comment:
The initial condition was:

```suggestion
    if (!(this.isActive && this.approvedAt) && !returnEvenIfNotApproved) {
```

but many tests are not setting `approvedAt` properly, and though I started fixing them there were just too many so I went for simplification.